homeassistant/components/knx/config_flow.py

```diff
@@ -164,37 +162,48 @@ async def async_step_manual_tunnel(
         self, user_input: dict | None = None
     ) -> FlowResult:
         """Manually configure tunnel connection parameters. Fields default to preselected gateway if one was found."""
+        errors: dict = {}
+
         if user_input is not None:
-            connection_type = user_input[CONF_KNX_TUNNELING_TYPE]
+            try:
+                _host = ip_v4_validator(user_input[CONF_HOST], multicast=False)
+            except vol.Invalid:
+                errors[CONF_HOST] = "invalid_ip_address"
+
+            if _local_ip := user_input.get(CONF_KNX_LOCAL_IP):
+                try:
+                    _local_ip = ip_v4_validator(_local_ip, multicast=False)
+                except vol.Invalid:
+                    errors[CONF_KNX_LOCAL_IP] = "invalid_ip_address"
+
+            if not errors:
+                connection_type = user_input[CONF_KNX_TUNNELING_TYPE]
+                entry_data = DEFAULT_ENTRY_DATA | KNXConfigEntryData(
+                    host=_host,
+                    port=user_input[CONF_PORT],
+                    route_back=(
+                        connection_type == CONF_KNX_LABEL_TUNNELING_UDP_ROUTE_BACK
+                    ),
+                    local_ip=_local_ip,
+                    connection_type=(
+                        CONF_KNX_TUNNELING_TCP
+                        if connection_type == CONF_KNX_LABEL_TUNNELING_TCP
+                        else CONF_KNX_TUNNELING
+                    ),
+                )
 
-            entry_data: KNXConfigEntryData = {
-                **DEFAULT_ENTRY_DATA,  # type: ignore[misc]
-                CONF_HOST: user_input[CONF_HOST],
-                CONF_PORT: user_input[CONF_PORT],
-                CONF_KNX_ROUTE_BACK: (
-                    connection_type == CONF_KNX_LABEL_TUNNELING_UDP_ROUTE_BACK
-                ),
-                CONF_KNX_LOCAL_IP: user_input.get(CONF_KNX_LOCAL_IP),
-                CONF_KNX_CONNECTION_TYPE: (
-                    CONF_KNX_TUNNELING_TCP
-                    if connection_type == CONF_KNX_LABEL_TUNNELING_TCP
-                    else CONF_KNX_TUNNELING
-                ),
-            }
+                if connection_type == CONF_KNX_LABEL_TUNNELING_TCP_SECURE:
+                    self._tunneling_config = entry_data
+                    return self.async_show_menu(
+                        step_id="secure_tunneling",
+                        menu_options=["secure_knxkeys", "secure_manual"],
+                    )
 
-            if connection_type == CONF_KNX_LABEL_TUNNELING_TCP_SECURE:
-                self._tunneling_config = entry_data
-                return self.async_show_menu(
-                    step_id="secure_tunneling",
-                    menu_options=["secure_knxkeys", "secure_manual"],
+                return self.async_create_entry(
+                    title=f"{CONF_KNX_TUNNELING.capitalize()} @ {_host}",
```

Comment:
```suggestion
                    title=f"Tunneling @ {_host}",
```

If we remove it from the other parts we should remove it here too